src/connection.cpp

```diff
@@ -132,15 +119,18 @@ std::vector<std::string> Connection::fetchInterfaces()
         if (std::strcmp(ifa->ifa_name, "lo") == 0)
             continue;
 
-        interfaces.emplace_back(ifa->ifa_name);
-    }
+        bool duplicate = false;
+        for (const auto& name : interfaces) {
+            if (std::strcmp(ifa->ifa_name, name.c_str()) == 0) {
+                duplicate = true;
+                break;
```

Comment:
I think it should be `continue` instead of `break`.

Reply:
That's right, my bad.